### PR TITLE
config: clarify xDS DiscoveryRequest behavior

### PR DESCRIPTION
**Pull request by fredlas**
Clarifies comments in discovery.proto about when response_nonce is(n't) set for (Delta)DiscoveryRequest. Also left a TODO in grpc_mux_impl_test.cc where spec-deviating behavior is expected.

Risk Level: none
Testing: comment-only change

### CODE REVIEW

File: api/envoy/api/v2/discovery.proto [hunk 1]
```diff
@@ -176,7 +177,7 @@ message DeltaDiscoveryRequest {
   // When the DeltaDiscoveryRequest is a ACK or NACK message in response
   // to a previous DeltaDiscoveryResponse, the response_nonce must be the
   // nonce in the DeltaDiscoveryResponse.
-  // Otherwise response_nonce must be omitted.
+  // Otherwise (unlike in DiscoveryRequest) response_nonce must be omitted.
```
Comment (lambdai) [inline api/envoy/api/v2/discovery.proto]: What are the possibility of `otherwise`? 
First request on the wire is an obvious one. What else?

Comment (fredlas) [reply api/envoy/api/v2/discovery.proto]: When the `DeltaDiscoveryRequest` is being used as an actual request, i.e., "here are names of resources I am newly (un)interested in". This is getting into what I believe @htuch has mentioned, that these (Delta)DiscoveryRequest messages are a mix of two functionalities that would perhaps be better as explicitly separate messages: informing the server of Envoy's interest in resources, and ACKing server messages. With such a separation, this sort of rule wouldn't even need to be mentioned.

Comment (htuch) [reply api/envoy/api/v2/discovery.proto]: Yeah, in UDPA we will make this separation. I think the clarifying comments are helpful in v2 incremental xDS, thanks @fredlas.

**Comment by htuch** [comment]
@fredlas can you merge master and potentially `fix_format`? I landed some major proto reformat PRs in the last day or so. Ta.

**Comment by fredlas** [comment]
Done.

**Pull request merged**